appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy

```diff
@@ -61,13 +72,17 @@ class AddDeviceTest extends AbstractAppServiceIntegrationTest
 		assert getResponseImmmediately.status == 200
 
 		when:
-		def response = appService.setNewDeviceRequest(richard.mobileNumber, "foo", "Some secret")
+		def responseWrongPassword = appService.setNewDeviceRequest(richard.mobileNumber, "foo", "Some secret")
+		def responseWrongMobileNumber = appService.setNewDeviceRequest("+31690609181", "foo", "Some secret")
 
 		then:
-		response.status == 400
+		responseWrongPassword.status == 400
+		responseWrongPassword.responseData.code == "error.decrypting.data"
```

Comment:
Would it not be more logical to have "error.wrong.mobile.number.or.password" in both cases?